src/components/CheckAuth/CheckAuth.js

```diff
@@ -0,0 +1,301 @@
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+import * as React from 'react';
+import { AppState } from 'react-native';
+import styled from 'styled-components/native';
+import { connect } from 'react-redux';
+import get from 'lodash.get';
+
+import type { Dispatch, RootReducerState } from 'reducers/rootReducer';
+import { DECRYPTING, INVALID_PASSWORD } from 'constants/walletConstants';
+import { checkAuthAction } from 'actions/authActions';
+import { Container, Wrapper } from 'components/Layout';
+import Loader from 'components/Loader';
+import ErrorMessage from 'components/ErrorMessage';
+import PinCode from 'components/PinCode';
+import { addAppStateChangeListener, removeAppStateChangeListener } from 'utils/common';
+import { getKeychainDataObject } from 'utils/keychain';
+import SlideModal from 'components/Modals/SlideModal';
+import Header from 'components/Header';
+import type { EthereumWallet } from 'models/Wallet';
+
+
+type HeaderProps = {
+  title?: String,
+  centerTitle?: boolean,
+  onClose?: Function,
+  onBack?: Function,
+};
+
+type ModalProps = {
+  onModalHide: Function,
+  onModalHidden?: Function,
+  title?: string,
+  centerTitle?: boolean,
+  fullScreen?: boolean,
+  showHeader?: boolean,
+  isVisible?: boolean,
+};
+
+type Props = {
+  checkPin: (pin: string, onValidPin: Function, options: Object) => void,
+  checkPrivateKey: (privateKey: string, onValidPin: Function) => void,
+  wallet: Object,
+  revealMnemonic: boolean,
+  onPinValid: Function,
+  isChecking: boolean,
+  title?: string,
+  useBiometrics: ?boolean,
+  enforcePin?: boolean,
+  modalProps?: ModalProps,
+  headerProps?: HeaderProps,
+  errorMessage?: string,
+};
+
+type State = {
+  biometricsShown: boolean,
+  lastAppState: string,
+  showPin: boolean,
+};
+
+type ValidPinCallback = (pin: string, wallet: EthereumWallet) => Promise<void>;
+
+const CheckAuthWrapper = styled(Container)`
+  margin-top: auto;
+  height: 100%;
+  flex: 1;
+`;
+
+const ACTIVE_APP_STATE = 'active';
+const BACKGROUND_APP_STATE = 'background';
+
+class CheckAuth extends React.Component<Props, State> {
+  static defaultProps = {
+    revealMnemonic: false,
+  };
+  state = {
+    biometricsShown: false,
+    lastAppState: AppState.currentState,
+    showPin: false,
+  };
+
+  componentDidMount() {
+    addAppStateChangeListener(this.handleAppStateChange);
+    const {
+      useBiometrics, revealMnemonic, enforcePin, modalProps,
+    } = this.props;
+    const { lastAppState } = this.state;
+
+    // do nothing if auth isn't supposed to be checked
+    if (modalProps && !get(modalProps, 'isVisible')) return;
+
+    if (useBiometrics
+      && !revealMnemonic
+      && lastAppState !== BACKGROUND_APP_STATE
+    ) {
+      this.showBiometricLogin();
+    } else if (lastAppState !== BACKGROUND_APP_STATE && !enforcePin) {
+      this.checkPrivateKey();
+    }
+  }
+
+  // special case for modals
+  componentDidUpdate = (prevProps: Props) => {
```

Comment:
Just to follow the pattern throughout components.
```suggestion
  componentDidUpdate(prevProps: Props) {
```